### PR TITLE
Changelog for v3.3.0

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+
+# Version 3.3.0
+
+_2022-05-04_
+
+This release brings WebSocket related improvements, support for HMPP, and a some performance improvements and bug fixes!
+
+## ✨️ [new] `WebSocketNetworkTransport.closeConnection` (#4049)
+
+This new method can be used in conjunction with [`reopenWhen`](https://apollographql.github.io/apollo-kotlin/kdoc/apollo-runtime/com.apollographql.apollo3/-apollo-client/-builder/web-socket-reopen-when.html)  to force a reconnection to the server. This could be useful for instance when needing to pass new auth tokens in the headers.
```
Comment (martinbonnin) [inline CHANGELOG.md:12]: Mention the link to 2.x `reconnect`? 

```
If you were using `subscriptionManager.reconnect()` in 2.x, `closeConnection` is a simple way to achieve the same behaviour.
```

And update the migration guide?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+
+## ✨️ [new] `GraphQLWsProtocol.connectionPayload` is now a lambda (#4043)
+
+With `GraphQLWsProtocol`, if you need to pass parameters to the connection payload, previously you would pass them as a static map to the builder. With this change you can now pass a lambda providing them as needed. This facilitates passing fresh auth tokens when connecting.
+
+## ✨️ [new] Hierarchical MultiPlatform Project (HMPP) (#4033)
+
+When using Apollo Kotlin on a multiplatform project, this release is compatible with the [hierarchical project structure](https://kotlinlang.org/docs/multiplatform-hierarchy.html), which makes it easier to share common code among several targets. This should also resolve [an issue](https://github.com/apollographql/apollo-kotlin/issues/4019) when using the library on a project using Kotlin 1.6.20+.
```
Comment (martinbonnin) [inline CHANGELOG.md:20]:
```suggestion
When using Apollo Kotlin on a multiplatform project, this release is compatible with the [hierarchical project structure](https://kotlinlang.org/docs/multiplatform-hierarchy.html), which makes it easier to share common code among several targets. Using HMPP in your project fixes errors some issues when compiling Kotlin metadata. See https://github.com/apollographql/apollo-kotlin/issues/4019 and https://youtrack.jetbrains.com/issue/KT-51970/ for more details.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+
+Many thanks to @CureleaAndrei and @kdk96 for contributing to this release! 🙏
+
+## ⚙️ Deprecations
+
+- `BearerTokenInterceptor` was provided as an example but is too simple for most use cases, and was therefore deprecated
+  in this release. [This page](https://www.apollographql.com/docs/kotlin/advanced/authentication) provides more details
+  about authentication.
+- The previous ways of passing parameters to the connection payload with `GraphQLWsProtocol` were deprecated (see above).
+
```
Comment (martinbonnin) [inline CHANGELOG.md:51]: I'm no english expert but I'd say use a progressive form?
```suggestion
- `BearerTokenInterceptor` was provided as an example but is too simple for most use cases, and has therefore been deprecated
  in this release. [This page](https://www.apollographql.com/docs/kotlin/advanced/authentication) provides more details
  about authentication.
- The previous ways of passing parameters to the connection payload with `GraphQLWsProtocol` has been deprecated (see above).

```

**Review by martinbonnin — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (BoD) [reply CHANGELOG.md:12]: (migration guide already pending [here](https://github.com/apollographql/apollo-kotlin/pull/4052/files?w=1))

**Pull request merged**